### PR TITLE
[New Integration] Export Indicators Service Integration

### PR DESCRIPTION
**Pull request by DeanArbel**
## Status
Ready

## Description
This integration provides Generic Outbound Indicators List as a service for the system indicators.

## Required version of Demisto
5.5.0

## Must have
- [x] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Leftovers:
https://github.com/demisto/etc/issues/21231

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](https://github.com/demisto/content/edit/generic-outbound/Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F5486)
- [ ] [CHANGELOG](https://github.com/demisto/content/edit/generic-outbound/Packs/ExportIndicators/Integrations/ExportIndicators/CHANGELOG.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F5486)
- [ ] [Readme](https://github.com/demisto/content/edit/generic-outbound/Packs/ExportIndicators/Integrations/ExportIndicators/README.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F5486)

### CODE REVIEW

File: Packs/GenericOutboundIndicators/Integrations/GenericOutboundIndicators/GenericOutboundIndicators.py [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+from CommonServerPython import *
+from CommonServerUserPython import *
+import json
+from flask import Flask, Response
+from gevent.pywsgi import WSGIServer
+from tempfile import NamedTemporaryFile
+from typing import Callable, List, Any
+
+''' GLOBAL VARIABLES '''
+INTEGRATION_NAME: str = 'Outbound Indicators Integration'
```
Comment (ronykoz) [inline Packs/GenericOutboundIndicators/Integrations/GenericOutboundIndicators/GenericOutboundIndicators.py]: Update the integration name as discussed

File: TestPlaybooks/playbook-Test_GenericOutboundIndicators.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
+id: Test_GOI
```
Comment (ronykoz) [inline TestPlaybooks/playbook-Test_GenericOutboundIndicators.yml]: Update the name of the test

File: Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.yml [hunk 3]
```diff
@@ -0,0 +1,125 @@
+category: Utilities
+commonfields:
+  id: ExportIndicators
+  version: -1
+configuration:
+- additionalinfo: The query to run to update its list. To view expected results, you can run the following command from the Demisto CLI
+    `!findIndicators query=<your query>`
```
Comment (ronykoz) [inline Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.yml]: was this changed as well?

File: Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.yml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  required: false
+  type: 0
+- defaultvalue: text
+  display: Outbound Format
+  hidden: false
+  name: format
+  options:
+  - text
+  - json
+  - json-seq
```
Comment (ronykoz) [inline Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.yml]: whats that?

File: Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.py [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+    Validates:
+        1. Valid port.
+        2. Valid cache_refresh_rate
+    """
+    get_params_port(params)
+    on_demand = params.get('on_demand', None)
+    if not on_demand:
+        try_parse_integer(params.get('list_size'), CTX_LIMIT_ERR_MSG)  # validate export_iocs Size was set
+        query = params.get('indicators_query')  # validate indicators_query isn't empty
+        if not query:
```
Comment (ronykoz) [inline Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.py]: if we are passing an empty query will it pass everything?

File: Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.py [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+
+def find_indicators_to_limit(indicator_query: str, limit: int) -> list:
+    """
+    Finds indicators using demisto.findIndicators
+    """
+    iocs, _ = find_indicators_to_limit_loop(indicator_query, limit)
+    return iocs[:limit]
+
+
+def find_indicators_to_limit_loop(indicator_query: str, limit: int, total_fetched: int = 0, next_page: int = 0,
```
Comment (ronykoz) [inline Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.py]: didn't understand the name of the function

File: Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.py [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+    return out_dict[CTX_VALUES_KEY]
+
+
+def save_context(now: datetime, out_dict: dict):
+    """Saves export_iocs state and refresh time to context"""
+    demisto.setLastRun({'last_run': date_to_timestamp(now)})
+    demisto.setIntegrationContext(out_dict)
+
+
+def find_indicators_to_limit(indicator_query: str, limit: int) -> list:
```
Comment (ronykoz) [inline Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.py]:
```suggestion
def find_indicators_with_limit(indicator_query: str, limit: int) -> list:
```

File: Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.yml (hunk 3 above)
Comment (DeanArbel) [reply Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.yml]: No, this wasn't changed

File: Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.yml (hunk 4 above)
Comment (DeanArbel) [reply Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.yml]: Render the list in JSON-SEQ format (https://tools.ietf.org/html/rfc7464) where each indicator basically is in JSON format on a single line, and there is no wrapping array. This format is useful for large feeds where the client do not have to load all the feed in memory but they can easily use a streaming parser.

File: Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.py (hunk 5 above)
Comment (DeanArbel) [reply Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.py]: Yeah

**Comment by DeanArbel** [comment]
> Update the img
> Is there a way where we can start the integration and pass the arguments/params in the URL itself?

Yeah, that should be pretty easy to do, but let's do it in another PR.

**Comment by ronykoz** [comment]
@DeanArbel let us know when to go over it

**Comment by DeanArbel** [comment]
@ronykoz it's ready for review

**Comment by kirbles19** [comment]
@DeanArbel Done.

**Comment by ronykoz** [comment]
@DeanArbel the build is not passing on the test you've added, please fix that and then we will merge

**Pull request merged**